### PR TITLE
[core] telemetry for when memory monitor is enabled

### PR DESCRIPTION
**Pull request by clarng**
Signed-off-by: Clarence Ng <clarence.wyng@gmail.com>

## Why are these changes needed?

Add telemetry to record at worker init time whether the ray cluster is using memory monitor

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Manual test per https://docs.google.com/document/d/1XSWzx9ZatnlttsurCXzvV4vVsb2g1KxJ2T4zpf33bI8/edit

cat /tmp/ray/session_latest/usage_stats.json

{"usage_stats": {"ray_version": "3.0.0.dev0", "python_version": "3.7.7", "schema_version": "0.1", "source": "OSS", "session_id": "330fafc5-718b-48c5-92d0-dbe8dbc784a5", "git_commit": "{{RAY_COMMIT_SHA}}", "os": "linux", "collect_timestamp_ms": 1665116028212, "session_start_timestamp_ms": 1665115658078, "cloud_provider": null, "min_workers": null, "max_workers": null, "head_node_instance_type": null, "worker_node_instance_types": null, "total_num_cpus": 12, "total_num_gpus": null, "total_memory_gb": 10.672229005023837, "total_object_store_memory_gb": 5.336114501580596, "library_usages": ["dataset", "train", "tune"], "total_success": 302, "total_failed": 0, "seq_number": 302, "extra_usage_tags": {"gcs_storage": "memory", "memory_monitor_enabled": "false"}, "total_num_nodes": 1, "total_num_running_jobs": 1}, "success": true, "error": null}

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
need fix CI though

**Review by rkooo567 — CHANGES_REQUESTED** [review]
Can we update this from usage_lib.py? I think we can do this way;

1. Generate a method that gets all internal config from `GetInternalConfigRequest`. 
2. When the dashboard first starts, we get internal config and update only relevant fields only once (in this case memory monitor config)

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -2117,6 +2117,15 @@ def connect(
             _setup_tracing()
             ray.__traced__ = True
 
+    # import here since record_extra_usage_tag depends on ray._private.worker
+    # that is initialized here.
+    from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
+
+    record_extra_usage_tag(
+        TagKey.MEMORY_MONITOR_ENABLED,
+        "true" if ray._raylet.Config.memory_monitor_interval_ms() > 0 else "false",
```
Comment (jjyao) [inline python/ray/_private/worker.py:2126]: Should we use the actual ms as the value, seems it provides more information while it's also not sensitive?

**Comment by rkooo567** [comment]
Hmm I think it is not going to work if you do ray start --head. Can you add a test with this case?

**Comment by clarng** [comment]
> I'd love to talk in person to talk about how I'd approach this

sounds good, should be back on this very soon

**Comment by clarng** [comment]
Fixed in https://github.com/ray-project/ray/pull/30472

**Pull request closed without merging**